public/app/features/plugins/admin/helpers.ts

```diff
@@ -425,3 +425,38 @@ export function filterByKeyword(plugins: CatalogPlugin[], query: string) {
   }
   return idxs.map((id) => getId(dataArray[id]));
 }
+
+export function isPluginUpdateable(plugin: CatalogPlugin) {
+  // If there is no update available, the plugin cannot be updated
+  if (!plugin.hasUpdate) {
+    return false;
+  }
+
+  // Provisioned plugins cannot be updated
+  if (plugin.isProvisioned) {
+    return false;
+  }
+
+  // Core plugins cannot be updated
+  if (plugin.isCore) {
+    return false;
+  }
+
+  // Currently renderer plugins are not supported by the catalog due to complications related to installation / update / uninstall.
+  if (plugin.type === PluginType.renderer) {
+    return false;
+  }
+
+  // Preinstalled plugins (with specified version) cannot be updated
```

Comment:
Is there a case when preinstalled plugins have `withVersion = false`?

Reply:
@andresmgot I think we want that case to be updateable, right?

Reply:
yeah, that's correct, if `isPreinstalled.withVersion` is false the version is not pinned and it can be updated.